### PR TITLE
add the interim gem5 conversion with ctest

### PR DESCRIPTION
**Pull request by jeffnye-gh**
GEM5 TAGE and SimpleBTB were converted to Sparta units, w/ Map style parameters and stats.

NOTE: some files are derived/modified versions, from external sources, please verify/confirm that the attached files have compatible licenses w/ riscv-perf-model intent.

this is a draft, next to zero cycles have been run on the conversion, basic sanity. The bpu is instantiated within a ctest bench. Not yet in the model proper.

The ideal outcome is this becomes one of many BPU's available to olympia.

Next steps might be to create local tests with training data, integrate with the existing BP API.

### CODE REVIEW

**Comment by klingaard** [comment]
This is really interesting and a good example of how to extract a Gem5 block.  Need to be careful of license agreements creeping into the code base.

File: test/core/bpu/gem5/src/base/debug.cpp [hunk 1]
```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2020 ARM Limited
+ * All rights reserved
+ *
```
[the remark this answers is not in the document]
Comment (jeffnye-gh) [reply test/core/bpu/gem5/src/base/debug.cpp:4]: this file is not used, I will remove from PR

File: test/core/bpu/gem5/src/base/debug.hpp [hunk 2]
```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2020 ARM Limited
+ * All rights reserved
+ *
+ * The license below extends only to copyright in the software and shall
```
[the remark this answers is not in the document]
Comment (jeffnye-gh) [reply test/core/bpu/gem5/src/base/debug.hpp:5]: this file is not used, I will remove from PR

**Comment by jeffnye-gh** [comment]
Those two files were the only problematic files.

File: test/core/bpu/commonTypes.h
[the remark this answers is not in the document]
Comment (jeffnye-gh) [reply test/core/bpu/commonTypes.h:1]: i have corrected the namespace in the latest push.

**Comment by jeffnye-gh** [comment]
I am going to close the PR, given the sensitivity to non-apache licenses having a gem5 conversion example seems problematic. This example could be done in a piece of documentation somewhere where the license issues are not a general concern.

I do plan to do a ground up implementation of various tage flavors, from the original literature, I will share that if possible.

**Pull request closed without merging**